+++
categories = ["computational-biology"]
coders = []
date = 2020-11-15T10:36:12+05:30
description = "Cancer Subtype Discovery from Multimodal Omic Data"
github = []
image = "https://image.flaticon.com/icons/svg/1472/1472090.svg"
title = "Cancer Subtyping Algorithm"
type = "post"

+++
# Cancer Subtype Discovery from Multi-modal Omic Data
Needless to state how bad a disease cancer is. [As per WHO](https://www.who.int/news-room/fact-sheets/detail/cancer), cancer is the second-largest non-infectious reason for the death by natural cause in humans. The earliest cancer research dates back to the 1700s. So it has been around for a very long time, but there is no promising cure for this disease yet. We have found the cure of so many diseases. Even completely eradicated many from the face of the earth. That makes it even stupefying that still there is no cure for cancer. The 21st century has witnessed medical progress sort of hitting a stumbling block. That is because unlike the disease whose treatment has been found, diseases like cancer simply do not respond to the conventional [one-size-fits-all solution](https://www.thepharmaletter.com/article/special-report-personalized-medicine). Cancer is not one disease but thousands of diseases in itself. Even the patients with the same type of cancer are [not necessarily the same at the molecular level](https://pubmed.ncbi.nlm.nih.gov/11173079/). That is why discovering a single drug to cure all cancer types is not possible as there is no single cure for cancer in the first place. It requires tailoring the medicinal treatment of a patient based on his/her molecular activity as manifested in the characteristics and symptoms during the treatment of a disease. AKA personalised treatment.

## Cancer Subtyping and Personalised Treatment
A decade or two back, diagnosis of cancer was made by analysing the pieces of the tumour under a microscope in a pathology lab[6]. That was merely by looking at the sample, the related conclusions about the disease were made. That includes cancer subtype, differentiation into malignant and benign, even judgement of its severity. It worked out for the most part but at the same time left much undiscovered. Today we know that it is caused by the aberration in the genes of the cells[7]. With [lowered cost of sequencing](https://www.genome.gov/about-genomics/fact-sheets/DNA-Sequencing-Costs-Data) since the past two decades, it has been made possible to sequence a large amount of genome from the cancerous tissues right away and strictly interrogate the individual tumour in order to hunt for the abnormal genes which caused cancer to progress. Earlier it was a common conviction that the only way to treat the disease cancer is through clinical trial and error. But the generation of a large amount of sequencing data had made it possible to come up with a high confidence hypothesis for the cancer subtype discovery much faster.

## Cancer Statistics and the Curse of Dimensionality
Once it was known that the gene aberration causes cancer, its diagnosis has turned from the tumour perspective to the molecular perspective. With lowered cost and ease of omic data generation for the cancerous cells, all the cancer databases are being overwhelmed by the unanalysed data. This brings along the common analytical problems with big data. One such problem is [“The Curse of Dimensionality”](https://en.wikipedia.org/wiki/Curse_of_dimensionality). Too many dimensions in the data lead to reduced accuracy in the results. Various analyses have found that the actual dimensions of the data can be much less than the observed dimensions. This contention between the actual dimensions and the observed dimensions is most probably subjected to the correlation in the data. In biological systems,  gene expression can be linked based on their expression mechanisms. This phenomenon is called the genetic correlation. For example, there are many polygenic traits in humans. Let’s take height as a trait. Then, in broad terms, it can be stated that because the trait- height is governed by so many genes, that there may be some other traits, say maximum running speed, which shows proportionality with the expression of height trait. So if we are analysing some data (not necessarily cancer data) which has features ‘Height’ and ‘Maximum speed’, then essentially both features are providing our machine learning model with the same information. So you might want to omit one of them to avoid redundancy.
In this project, an attempt has been made to reduce the dimensionality of the data while keeping as much cluster information as possible.

---
*That was little about the project's deal. After the project tenure was over, I continued my work on the algorithm under Dr Sushmita Paul in the CB lab, IIT Jodhpur. However, I can't discuss the algorithm in this post until it's published (soon enough!).*
